```
[ERROR] …/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java:165: error: reference not found
[ERROR] * Standard implementations of {@link ParameterizedJob#isParametrized}.
[ERROR] ^
[ERROR] 
```

--- a/core/src/main/java/hudson/model/AbstractProject.java
+++ b/core/src/main/java/hudson/model/AbstractProject.java
@@ -464,7 +464,7 @@ public abstract class AbstractProject<P extends AbstractProject<P,R>,R extends A
      */
     public String getBuildNowText() {
         // For compatibility, still use the deprecated replacer if specified.
-        return AlternativeUiTextProvider.get(BUILD_NOW_TEXT, this, getParameterizedJobMixIn().getBuildNowText());
+        return AlternativeUiTextProvider.get(BUILD_NOW_TEXT, this, ParameterizedJobMixIn.ParameterizedJob.super.getBuildNowText());
     }
 
     /**
@@ -832,14 +832,13 @@ public abstract class AbstractProject<P extends AbstractProject<P,R>,R extends A
      *      For the convenience of the caller, this collection can contain null, and those will be silently ignored.
      * @since 1.383
      */
-    @SuppressWarnings("unchecked")
     @WithBridgeMethods(Future.class)
     public QueueTaskFuture<R> scheduleBuild2(int quietPeriod, Cause c, Collection<? extends Action> actions) {
         List<Action> queueActions = new ArrayList<Action>(actions);
         if (c != null) {
             queueActions.add(new CauseAction(c));
         }
-        return getParameterizedJobMixIn().scheduleBuild2(quietPeriod, queueActions.toArray(new Action[queueActions.size()]));
+        return scheduleBuild2(quietPeriod, queueActions.toArray(new Action[queueActions.size()]));
     }
 
     /**
@@ -863,6 +862,11 @@ public abstract class AbstractProject<P extends AbstractProject<P,R>,R extends A
     @WithBridgeMethods(Future.class)
     public QueueTaskFuture<R> scheduleBuild2(int quietPeriod, Cause c) {
         return scheduleBuild2(quietPeriod, c, new Action[0]);
+    }
+
+    @Override
+    public QueueTaskFuture<R> scheduleBuild2(int quietPeriod, Action... actions) {
+        return ParameterizedJobMixIn.ParameterizedJob.super.scheduleBuild2(quietPeriod, actions);
     }
 
     /**

--- a/core/src/main/java/hudson/model/Project.java
+++ b/core/src/main/java/hudson/model/Project.java
@@ -26,7 +26,6 @@ package hudson.model;
 
 import hudson.Util;
 import hudson.model.Descriptor.FormException;
-import hudson.model.queue.QueueTaskFuture;
 import hudson.scm.SCM;
 import hudson.tasks.BuildStep;
 import hudson.tasks.BuildWrapper;
@@ -107,10 +106,6 @@ public abstract class Project<P extends Project<P,B>,B extends Build<P,B>>
 
     @Override public Item asItem() {
         return this;
-    }
-
-    @Override public QueueTaskFuture<?> scheduleBuild2(int quietPeriod, Action... actions) {
-        return scheduleBuild2(quietPeriod, null, actions);
     }
 
     @Override public SCMTrigger getSCMTrigger() {

--- a/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java
+++ b/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java
@@ -162,7 +162,7 @@ public abstract class ParameterizedJobMixIn<JobT extends Job<JobT, RunT> & Param
     }
 
     /**
-     * Standard implementations of {@link ParameterizedJob#isParametrized}.
+     * Standard implementations of {@link ParameterizedJob#isParameterized}.
      */
     public final boolean isParameterized() {
         return asJob().getProperty(ParametersDefinitionProperty.class) != null;

--- a/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java
+++ b/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java
@@ -66,6 +66,7 @@ import org.kohsuke.stapler.interceptor.RequirePOST;
 /**
  * Allows a {@link Job} to make use of {@link ParametersDefinitionProperty} and be scheduled in various ways.
  * Stateless so there is no need to keep an instance of it in a field.
+ * Besides implementing {@link ParameterizedJob}, you should override {@link Job#makeSearchIndex} to call {@link #extendSearchIndex}.
  * @since 1.556
  */
 @SuppressWarnings("unchecked") // AbstractItem.getParent does not correctly override; scheduleBuild2 inherently untypable
@@ -96,11 +97,7 @@ public abstract class ParameterizedJobMixIn<JobT extends Job<JobT, RunT> & Param
     }
 
     /**
-     * Provides a standard implementation of {@link SCMTriggerItem#scheduleBuild2} to schedule a build with the ability to wait for its result.
-     * That job method is often used during functional tests ({@code JenkinsRule.assertBuildStatusSuccess}).
-     * @param quietPeriod seconds to wait before starting (normally 0)
-     * @param actions various actions to associate with the scheduling, such as {@link ParametersAction} or {@link CauseAction}
-     * @return a handle by which you may wait for the build to complete (or just start); or null if the build was not actually scheduled for some reason
+     * Standard implementation of {@link ParameterizedJob#scheduleBuild2}.
      */
     public final @CheckForNull QueueTaskFuture<RunT> scheduleBuild2(int quietPeriod, Action... actions) {
         Queue.Item i = scheduleBuild2(quietPeriod, Arrays.asList(actions));
@@ -162,7 +159,7 @@ public abstract class ParameterizedJobMixIn<JobT extends Job<JobT, RunT> & Param
     }
 
     /**
-     * Standard implementations of {@link ParameterizedJob#isParameterized}.
+     * Standard implementation of {@link ParameterizedJob#isParameterized}.
      */
     public final boolean isParameterized() {
         return asJob().getProperty(ParametersDefinitionProperty.class) != null;
@@ -372,9 +369,17 @@ public abstract class ParameterizedJobMixIn<JobT extends Job<JobT, RunT> & Param
             return getParameterizedJobMixIn().scheduleBuild(quietPeriod, c);
         }
 
-        // omitting scheduleBuild2(int, Action...) since it is defined in SCMTriggerItem (could include less commonly used overloads if desired)
-
-        // cannot offer makeSearchIndex() since it is defined in Job
+        /**
+         * Provides a standard implementation of {@link SCMTriggerItem#scheduleBuild2} to schedule a build with the ability to wait for its result.
+         * That job method is often used during functional tests ({@code JenkinsRule.assertBuildStatusSuccess}).
+         * @param quietPeriod seconds to wait before starting (normally 0)
+         * @param actions various actions to associate with the scheduling, such as {@link ParametersAction} or {@link CauseAction}
+         * @return a handle by which you may wait for the build to complete (or just start); or null if the build was not actually scheduled for some reason
+         */
+        @CheckForNull
+        default QueueTaskFuture<RunT> scheduleBuild2(int quietPeriod, Action... actions) {
+            return getParameterizedJobMixIn().scheduleBuild2(quietPeriod, actions);
+        }
 
         /**
          * Schedules a new build command.

--- a/core/src/main/java/jenkins/triggers/SCMTriggerItem.java
+++ b/core/src/main/java/jenkins/triggers/SCMTriggerItem.java
@@ -57,7 +57,7 @@ public interface SCMTriggerItem {
     /** @see jenkins.model.ParameterizedJobMixIn.ParameterizedJob#getQuietPeriod */
     int getQuietPeriod();
 
-    /** @see ParameterizedJobMixIn#scheduleBuild2 */
+    /** @see jenkins.model.ParameterizedJobMixIn.ParameterizedJob#scheduleBuild2 */
     @CheckForNull QueueTaskFuture<?> scheduleBuild2(int quietPeriod, Action... actions);
 
     /**

--- a/pom.xml
+++ b/pom.xml
@@ -467,7 +467,7 @@ THE SOFTWARE.
         <plugin>
           <groupId>com.infradna.tool</groupId>
           <artifactId>bridge-method-injector</artifactId>
-          <version>1.15</version>
+          <version>1.16-20170502.210203-1</version> <!-- TODO https://github.com/infradna/bridge-method-injector/pull/15 -->
         </plugin>
         <plugin>
           <groupId>org.codehaus.mojo</groupId>

